scipy/spatial/src/distance_impl.h

```diff
@@ -167,26 +167,27 @@ yule_bool_distance_char(const char *u, const char *v, npy_intp n)
     npy_intp ntt = 0, nff = 0, nft = 0, ntf = 0;
 
     for (i = 0; i < n; i++) {
-        ntt += (u[i] && v[i]);
-        ntf += (u[i] && !v[i]);
-        nft += (!u[i] && v[i]);
-        nff += (!u[i] && !v[i]);
+        npy_bool x = u[i], y = v[i];
+        ntt += x & y;
+        ntf += x & (!y);
+        nft += (!x) & y;
```

Comment:
The assumption is that only the low-order bit is used. The code that produces the boolean arrays uses `if X.dtype != bool: X = X.astype(bool)`; shouldn't that produce proper booleans?

(I guess the type should be `npy_bool`, not `char`. I was just doing a big refactor, can take that into account.)